components/camel-yql/src/main/java/org/apache/camel/component/yql/YqlComponent.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.yql;
+
+import java.util.Map;
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.yql.configuration.YqlConfiguration;
+import org.apache.camel.component.yql.configuration.YqlConfigurationValidator;
+import org.apache.camel.impl.DefaultComponent;
+
+public class YqlComponent extends DefaultComponent {
+
+    @Override
+    protected Endpoint createEndpoint(final String uri, final String remaining, final Map<String, Object> parameters) throws Exception {
+        final YqlConfiguration configuration = new YqlConfiguration();
+        configuration.setQuery(remaining);
+        configuration.setFormat((String) parameters.getOrDefault("format", "json"));
```

Comment:
Camel can do this automatic in the setProperties(configuration, parameters) you do further below. If you want to set default values then set them on the configuration class instead

Reply:
Done